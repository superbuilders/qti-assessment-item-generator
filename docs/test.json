{
	"identifier": "number-line-distance-0-4",
	"title": "Identify 0.4 on a Number Line",
	"responseDeclarations": [
		{
			"identifier": "RESPONSE",
			"cardinality": "single",
			"baseType": "identifier",
			"correct": "C"
		}
	],
	"body": [
		{
			"type": "interactionRef",
			"interactionId": "choice_interaction"
		}
	],
	"interactions": {
		"choice_interaction": {
			"type": "choiceInteraction",
			"responseIdentifier": "RESPONSE",
			"prompt": [
				{
					"type": "text",
					"content": "Which number line shows a point that represents a distance of "
				},
				{
					"type": "math",
					"mathml": "<mn>0.4</mn>"
				},
				{
					"type": "text",
					"content": " units from zero? Select one answer."
				}
			],
			"choices": [
				{
					"identifier": "A",
					"content": [
						{
							"type": "widgetRef",
							"widgetId": "repl__choice_interaction__choice__a__visual__0",
							"widgetType": "numberLineWithAction"
						}
					]
				},
				{
					"identifier": "B",
					"content": [
						{
							"type": "widgetRef",
							"widgetId": "repl__choice_interaction__choice__b__visual__0",
							"widgetType": "numberLineWithAction"
						}
					]
				},
				{
					"identifier": "C",
					"content": [
						{
							"type": "widgetRef",
							"widgetId": "repl__choice_interaction__choice__c__visual__0",
							"widgetType": "numberLineWithAction"
						}
					]
				},
				{
					"identifier": "D",
					"content": [
						{
							"type": "widgetRef",
							"widgetId": "repl__choice_interaction__choice__d__visual__0",
							"widgetType": "numberLineWithAction"
						}
					]
				}
			],
			"shuffle": true,
			"minChoices": 1,
			"maxChoices": 1
		}
	},
	"widgets": {
		"repl__choice_interaction__choice__a__visual__0": {
			"type": "numberLineWithAction",
			"width": 338,
			"height": 320,
			"orientation": "horizontal",
			"min": 0,
			"max": 1,
			"tickInterval": {
				"type": "whole",
				"interval": 1
			},
			"secondaryTickInterval": {
				"type": "fraction",
				"numerator": 1,
				"denominator": 10
			},
			"startValue": 0,
			"actions": [
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				}
			]
		},
		"repl__choice_interaction__choice__b__visual__0": {
			"type": "numberLineWithAction",
			"width": 338,
			"height": 320,
			"orientation": "horizontal",
			"min": 0,
			"max": 1,
			"tickInterval": {
				"type": "whole",
				"interval": 1
			},
			"secondaryTickInterval": {
				"type": "fraction",
				"numerator": 1,
				"denominator": 10
			},
			"startValue": 0,
			"actions": [
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				}
			]
		},
		"repl__choice_interaction__choice__c__visual__0": {
			"type": "numberLineWithAction",
			"width": 338,
			"height": 320,
			"orientation": "horizontal",
			"min": 0,
			"max": 1,
			"tickInterval": {
				"type": "whole",
				"interval": 1
			},
			"secondaryTickInterval": {
				"type": "fraction",
				"numerator": 1,
				"denominator": 10
			},
			"startValue": 0,
			"actions": [
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				}
			]
		},
		"repl__choice_interaction__choice__d__visual__0": {
			"type": "numberLineWithAction",
			"width": 338,
			"height": 320,
			"orientation": "horizontal",
			"min": 0,
			"max": 1,
			"tickInterval": {
				"type": "whole",
				"interval": 1
			},
			"secondaryTickInterval": {
				"type": "fraction",
				"numerator": 1,
				"denominator": 10
			},
			"startValue": 0,
			"actions": [
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				},
				{
					"delta": {
						"type": "fraction",
						"numerator": 1,
						"denominator": 10,
						"sign": "+"
					}
				}
			]
		}
	},
	"feedbackPlan": {
		"mode": "combo",
		"dimensions": [
			{
				"responseIdentifier": "RESPONSE",
				"kind": "enumerated",
				"keys": ["A", "B", "C", "D"]
			}
		],
		"combinations": [
			{
				"id": "FB__RESPONSE_A",
				"path": [
					{
						"responseIdentifier": "RESPONSE",
						"key": "A"
					}
				]
			},
			{
				"id": "FB__RESPONSE_B",
				"path": [
					{
						"responseIdentifier": "RESPONSE",
						"key": "B"
					}
				]
			},
			{
				"id": "FB__RESPONSE_C",
				"path": [
					{
						"responseIdentifier": "RESPONSE",
						"key": "C"
					}
				]
			},
			{
				"id": "FB__RESPONSE_D",
				"path": [
					{
						"responseIdentifier": "RESPONSE",
						"key": "D"
					}
				]
			}
		]
	},
	"feedback": {
		"FEEDBACK__OVERALL": {
			"RESPONSE": {
				"A": {
					"content": {
						"preamble": {
							"correctness": "incorrect",
							"summary": [
								{
									"type": "text",
									"content": "It looks like the selected number line does not place the point at "
								},
								{
									"type": "math",
									"mathml": "<mn>0.4</mn>"
								},
								{
									"type": "text",
									"content": " units from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": ", which is "
								},
								{
									"type": "math",
									"mathml": "<mfrac><mn>4</mn><mn>10</mn></mfrac>"
								},
								{
									"type": "text",
									"content": " of the way from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": " to "
								},
								{
									"type": "math",
									"mathml": "<mn>1</mn>"
								},
								{
									"type": "text",
									"content": "."
								}
							]
						},
						"steps": [
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Interpret "
									},
									{
										"type": "math",
										"mathml": "<mn>0.4</mn>"
									},
									{
										"type": "text",
										"content": " as four tenths."
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Recall that each tenth is "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": " unit."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "So "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " equals "
											},
											{
												"type": "math",
												"mathml": "<mfrac><mn>4</mn><mn>10</mn></mfrac>"
											},
											{
												"type": "text",
												"content": ", meaning four jumps of size "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": " from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "It lies between "
											},
											{
												"type": "math",
												"mathml": "<mn>0.3</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": ", and "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn><mo>&lt;</mo><mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": ", so it is a little less than halfway from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Use the scale of the number line."
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Look for a number line labeled from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": " that is divided into ten equal parts; each small tick then represents "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Starting at "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", count "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " equal ticks to the right; that location corresponds to "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Self-check before selecting."
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Does the segment from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": " have ten equal parts (tenths)?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Is the marked point exactly "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " of those equal parts to the right of "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", and to the left of "
											},
											{
												"type": "math",
												"mathml": "<mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": " since "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn><mo>&lt;</mo><mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									}
								]
							}
						]
					}
				},
				"B": {
					"content": {
						"preamble": {
							"correctness": "incorrect",
							"summary": [
								{
									"type": "text",
									"content": "The point should be exactly "
								},
								{
									"type": "math",
									"mathml": "<mn>0.4</mn>"
								},
								{
									"type": "text",
									"content": " units from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": ", which is four tenths; the number line you selected does not place the point that distance from zero."
								}
							]
						},
						"steps": [
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Determine the size of one tick mark"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Look between "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": ". Count the equal spaces. If there are "
											},
											{
												"type": "math",
												"mathml": "<mn>10</mn>"
											},
											{
												"type": "text",
												"content": ", then each space is "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": " (one tenth)."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Four tick marks from zero represent "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " tenths: "
											},
											{
												"type": "math",
												"mathml": "<mfrac><mn>4</mn><mn>10</mn></mfrac>"
											},
											{
												"type": "text",
												"content": " which equals "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Use distance from zero"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Distance ignores direction: points at "
											},
											{
												"type": "math",
												"mathml": "<mo>+</mo><mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mo>−</mo><mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " are both a distance of "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " because "
											},
											{
												"type": "math",
												"mathml": "<mo>|</mo><mi>x</mi><mo>|</mo>"
											},
											{
												"type": "text",
												"content": " measures how far "
											},
											{
												"type": "math",
												"mathml": "<mi>x</mi>"
											},
											{
												"type": "text",
												"content": " is from zero."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Starting at "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", count "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " tick marks of size "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": " to locate the point. It should land exactly on the 4th tick, not past it."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Self-check and revise your choice"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Ask yourself:"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "• Do you see ten equal spaces between "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "• Is the point exactly the 4th tick from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " (to the right or to the left), not at "
											},
											{
												"type": "math",
												"mathml": "<mn>0.04</mn>"
											},
											{
												"type": "text",
												"content": " or at "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Use these checks to select the number line that shows a point exactly "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " units from zero."
											}
										]
									}
								]
							}
						]
					}
				},
				"C": {
					"content": {
						"preamble": {
							"correctness": "correct",
							"summary": [
								{
									"type": "text",
									"content": " "
								},
								{
									"type": "math",
									"mathml": "<mn>0.4</mn>"
								},
								{
									"type": "text",
									"content": " represents four tenths of a unit from zero. On a number line from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": " to "
								},
								{
									"type": "math",
									"mathml": "<mn>1</mn>"
								},
								{
									"type": "text",
									"content": " split into ten equal parts, this is the fourth tick to the right of zero."
								}
							]
						},
						"steps": [
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Interpret the decimal as tenths"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Recall that "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn><mo>=</mo><mfrac><mn>4</mn><mn>10</mn></mfrac>"
											},
											{
												"type": "text",
												"content": ", which means four tenths of one whole unit."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Locate the point on a 0–1 number line"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "If the interval from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": " is divided into ten equal segments, each tick represents "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Starting at zero, move "
											},
											{
												"type": "math",
												"mathml": "<mn>4</mn>"
											},
											{
												"type": "text",
												"content": " ticks to the right to land at "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Connect to your selection"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "The diagram you chose places the point exactly four tenths of a unit from zero, matching "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									}
								]
							}
						]
					}
				},
				"D": {
					"content": {
						"preamble": {
							"correctness": "incorrect",
							"summary": [
								{
									"type": "text",
									"content": "The diagram you chose does not show a point exactly four tenths of a unit from zero."
								},
								{
									"type": "text",
									"content": " On a number line from "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": " to "
								},
								{
									"type": "math",
									"mathml": "<mn>1</mn>"
								},
								{
									"type": "text",
									"content": " split into ten equal parts, each tick represents "
								},
								{
									"type": "math",
									"mathml": "<mn>0.1</mn>"
								},
								{
									"type": "text",
									"content": ", so "
								},
								{
									"type": "math",
									"mathml": "<mn>0.4</mn>"
								},
								{
									"type": "text",
									"content": " is at the fourth tick to the right of "
								},
								{
									"type": "math",
									"mathml": "<mn>0</mn>"
								},
								{
									"type": "text",
									"content": "."
								}
							]
						},
						"steps": [
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Interpret the distance"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "“Distance from zero” means how far the point is from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", not which direction."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Think of "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": " as four tenths."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Use tenths on the 0–1 number line"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Check that the segment from "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " to "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": " is divided into ten equal spaces; each space is "
											},
											{
												"type": "math",
												"mathml": "<mn>0.1</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Starting at "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", count four equal intervals to the right. That location corresponds to "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "."
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Avoid common slips like stopping at the fifth tick ("
											},
											{
												"type": "math",
												"mathml": "<mn>0.5</mn>"
											},
											{
												"type": "text",
												"content": ") or using uneven partitions."
											}
										]
									}
								]
							},
							{
								"type": "step",
								"title": [
									{
										"type": "text",
										"content": "Self-check and revise your choice"
									}
								],
								"content": [
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Ask yourself:"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "• Does the diagram have exactly ten equal spaces between "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": " and "
											},
											{
												"type": "math",
												"mathml": "<mn>1</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "• Is the point exactly four tick marks to the right of "
											},
											{
												"type": "math",
												"mathml": "<mn>0</mn>"
											},
											{
												"type": "text",
												"content": ", representing "
											},
											{
												"type": "math",
												"mathml": "<mn>0.4</mn>"
											},
											{
												"type": "text",
												"content": "?"
											}
										]
									},
									{
										"type": "paragraph",
										"content": [
											{
												"type": "text",
												"content": "Use these checks to select the diagram that matches this distance."
											}
										]
									}
								]
							}
						]
					}
				}
			}
		}
	}
}
