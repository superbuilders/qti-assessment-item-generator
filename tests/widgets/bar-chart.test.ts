import { describe, expect, test } from "bun:test"
import * as errors from "@superbuilders/errors"
import * as logger from "@superbuilders/slog"
import { barChartExamples } from "../../examples/bar-chart"
import type { WidgetInput } from "../../src/widgets/registry"
import { generateWidgetLegacy } from "../../src/widgets/widget-generator"

describe("Widget: bar-chart", () => {
	const examples: WidgetInput[] = barChartExamples

	examples.forEach((props, index) => {
		test(`should produce consistent output for example #${index + 1}`, async () => {
			const result = await errors.try(generateWidgetLegacy(props))
			if (result.error) {
				logger.error("widget generation failed", { error: result.error, index })
				throw result.error
			}
			expect(result.data).toMatchSnapshot()
		})
	})
})
