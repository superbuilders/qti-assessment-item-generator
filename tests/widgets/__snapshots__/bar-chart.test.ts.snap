// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget: bar-chart should produce consistent output for example #1 1`] = `"<svg width="568" height="457" viewBox="-80 -45 568 457" xmlns="http://www.w3.org/2000/svg" font-family="sans-serif" font-size="12"><defs><clipPath id="clip-0"><rect x="0" y="0" width="480" height="340"/></clipPath></defs><text x="240" y="-36.599999999999994" font-size="18" text-anchor="middle" fill="#333333" dominant-baseline="hanging"><tspan x="240" dy="0em">Games Won Last Summer</tspan></text><line x1="0" y1="0" x2="0" y2="340" stroke="#333333" stroke-width="1.5"/><line x1="-5" y1="340" x2="0" y2="340" stroke="#333333" stroke-width="1.5"/><text x="-8" y="344" font-size="12" text-anchor="end" fill="#333333">0</text><line x1="0" y1="340" x2="480" y2="340" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="297.5" x2="0" y2="297.5" stroke="#333333" stroke-width="1.5"/><text x="-8" y="301.5" font-size="12" text-anchor="end" fill="#333333">2</text><line x1="0" y1="297.5" x2="480" y2="297.5" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="255" x2="0" y2="255" stroke="#333333" stroke-width="1.5"/><text x="-8" y="259" font-size="12" text-anchor="end" fill="#333333">4</text><line x1="0" y1="255" x2="480" y2="255" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="212.5" x2="0" y2="212.5" stroke="#333333" stroke-width="1.5"/><text x="-8" y="216.5" font-size="12" text-anchor="end" fill="#333333">6</text><line x1="0" y1="212.5" x2="480" y2="212.5" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="170" x2="0" y2="170" stroke="#333333" stroke-width="1.5"/><text x="-8" y="174" font-size="12" text-anchor="end" fill="#333333">8</text><line x1="0" y1="170" x2="480" y2="170" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="127.5" x2="0" y2="127.5" stroke="#333333" stroke-width="1.5"/><text x="-8" y="131.5" font-size="12" text-anchor="end" fill="#333333">10</text><line x1="0" y1="127.5" x2="480" y2="127.5" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="85" x2="0" y2="85" stroke="#333333" stroke-width="1.5"/><text x="-8" y="89" font-size="12" text-anchor="end" fill="#333333">12</text><line x1="0" y1="85" x2="480" y2="85" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="42.5" x2="0" y2="42.5" stroke="#333333" stroke-width="1.5"/><text x="-8" y="46.5" font-size="12" text-anchor="end" fill="#333333">14</text><line x1="0" y1="42.5" x2="480" y2="42.5" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="0" x2="0" y2="0" stroke="#333333" stroke-width="1.5"/><text x="-8" y="4" font-size="12" text-anchor="end" fill="#333333">16</text><line x1="0" y1="0" x2="480" y2="0" stroke="#e0e0e0" stroke-width="1"/><text x="-61.6" y="170" font-size="16" text-anchor="middle" fill="#333333" dominant-baseline="middle" transform="rotate(-90, -61.6, 170)"><tspan x="-61.6" dy="0em">Games Won</tspan></text><line x1="0" y1="340" x2="480" y2="340" stroke="#333333" stroke-width="1.5"/><line x1="80" y1="340" x2="80" y2="345" stroke="#333333" stroke-width="1.5"/><text x="80" y="353" font-size="12" text-anchor="middle" dominant-baseline="hanging" fill="#333333">Lions</text><line x1="240" y1="340" x2="240" y2="345" stroke="#333333" stroke-width="1.5"/><text x="240" y="353" font-size="12" text-anchor="middle" dominant-baseline="hanging" fill="#333333">Tigers</text><line x1="400" y1="340" x2="400" y2="345" stroke="#333333" stroke-width="1.5"/><text x="400" y="353" font-size="12" text-anchor="middle" dominant-baseline="hanging" fill="#333333">Bears</text><text x="240" y="387" font-size="16" text-anchor="middle" fill="#333333"><tspan x="240" dy="0em">Team</tspan></text><g clip-path="url(#clip-0)"><rect x="16" y="42.5" width="128" height="297.5" fill="#4285F4"/><rect x="176" y="297.5" width="128" height="42.5" fill="#4285F4"/><rect x="336" y="191.25" width="128" height="148.75" fill="#4285F4"/></g></svg>"`;

exports[`Widget: bar-chart should produce consistent output for example #2 1`] = `"<svg width="488" height="417" viewBox="-80 -45 488 417" xmlns="http://www.w3.org/2000/svg" font-family="sans-serif" font-size="12"><defs><clipPath id="clip-0"><rect x="0" y="0" width="400" height="300"/></clipPath></defs><text x="200" y="-36.599999999999994" font-size="18" text-anchor="middle" fill="#333333" dominant-baseline="hanging"><tspan x="200" dy="0em">Favorite Ice Cream Flavors</tspan></text><line x1="0" y1="0" x2="0" y2="300" stroke="#333333" stroke-width="1.5"/><line x1="-5" y1="300" x2="0" y2="300" stroke="#333333" stroke-width="1.5"/><text x="-8" y="304" font-size="12" text-anchor="end" fill="#333333">0</text><line x1="0" y1="300" x2="400" y2="300" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="240" x2="0" y2="240" stroke="#333333" stroke-width="1.5"/><text x="-8" y="244" font-size="12" text-anchor="end" fill="#333333">5</text><line x1="0" y1="240" x2="400" y2="240" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="180" x2="0" y2="180" stroke="#333333" stroke-width="1.5"/><text x="-8" y="184" font-size="12" text-anchor="end" fill="#333333">10</text><line x1="0" y1="180" x2="400" y2="180" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="120" x2="0" y2="120" stroke="#333333" stroke-width="1.5"/><text x="-8" y="124" font-size="12" text-anchor="end" fill="#333333">15</text><line x1="0" y1="120" x2="400" y2="120" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="60" x2="0" y2="60" stroke="#333333" stroke-width="1.5"/><text x="-8" y="64" font-size="12" text-anchor="end" fill="#333333">20</text><line x1="0" y1="60" x2="400" y2="60" stroke="#e0e0e0" stroke-width="1"/><line x1="-5" y1="0" x2="0" y2="0" stroke="#333333" stroke-width="1.5"/><text x="-8" y="4" font-size="12" text-anchor="end" fill="#333333">25</text><line x1="0" y1="0" x2="400" y2="0" stroke="#e0e0e0" stroke-width="1"/><text x="-61.6" y="150" font-size="16" text-anchor="middle" fill="#333333" dominant-baseline="middle" transform="rotate(-90, -61.6, 150)"><tspan x="-61.6" dy="0em">Number of Votes</tspan></text><line x1="0" y1="300" x2="400" y2="300" stroke="#333333" stroke-width="1.5"/><line x1="50" y1="300" x2="50" y2="305" stroke="#333333" stroke-width="1.5"/><text x="50" y="313" font-size="12" text-anchor="middle" dominant-baseline="hanging" fill="#333333">Vanilla</text><line x1="150" y1="300" x2="150" y2="305" stroke="#333333" stroke-width="1.5"/><text x="150" y="313" font-size="12" text-anchor="middle" dominant-baseline="hanging" fill="#333333">Chocolate</text><line x1="250" y1="300" x2="250" y2="305" stroke="#333333" stroke-width="1.5"/><text x="250" y="313" font-size="12" text-anchor="middle" dominant-baseline="hanging" fill="#333333">Strawberry</text><line x1="350" y1="300" x2="350" y2="305" stroke="#333333" stroke-width="1.5"/><text x="350" y="313" font-size="12" text-anchor="middle" dominant-baseline="hanging" fill="#333333">Mint</text><text x="200" y="347" font-size="16" text-anchor="middle" fill="#333333"><tspan x="200" dy="0em">Flavor</tspan></text><g clip-path="url(#clip-0)"><rect x="10" y="84" width="80" height="216" fill="#FF6B6B"/><rect x="110" y="36" width="80" height="264" fill="#FF6B6B"/><rect x="210" y="156" width="80" height="144" fill="#FF6B6B"/><rect x="310" y="204" width="80" height="96" fill="#FF6B6B"/></g></svg>"`;
